### DISCUSSION TITLE
Why CIL doesn't have ART and SART algorithm?

### DISCUSSION THREAD

**Discussion by TianSong1991** [TomographicImaging/CIL-Demos#137] [Q&A]
Hello, I want to know why CIL doesn't have ART and SART algorithm. Will the CIL have ART and SART algorithm in the future?

**Comment by MargaretDuff** [comment]
Hi @TianSong1991! Thanks for your message and for looking into CIL. We currently have a number of algorithms for iterative reconstruction in CIL, including ADMM, CGLS, FISTA, GD, PDHG, SIRT and SPDHG. Perhaps SIRT will do what you are looking for? If you give us some more details as to the problem you wish to solve, we may also be able to advise a suitable approach. 

We are currently focused on stochastic algorithm development: balancing the needs of our mathematical and CT user communities with limited developer time. You are of course welcome to write your own algorithm, subclassing the algorithm class (https://github.com/TomographicImaging/CIL/blob/master/Wrappers/Python/cil/optimisation/algorithms/Algorithm.py) . If you do so, please do submit a pull request and we can add your contribution to CIL!
